x-pack/plugins/security_solution/server/endpoint/services/actions/mocks.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import { ElasticsearchClientMock } from '@kbn/core/server/mocks';
+import { AGENT_ACTIONS_INDEX, AGENT_ACTIONS_RESULTS_INDEX } from '@kbn/fleet-plugin/common';
+import { EndpointActionGenerator } from '../../../../common/endpoint/data_generators/endpoint_action_generator';
+import { FleetActionGenerator } from '../../../../common/endpoint/data_generators/fleet_action_generator';
+import {
+  EndpointAction,
+  EndpointActionResponse,
+  LogsEndpointAction,
+  LogsEndpointActionResponse,
+} from '../../../../common/endpoint/types';
+import {
+  ENDPOINT_ACTION_RESPONSES_INDEX_PATTERN,
+  ENDPOINT_ACTIONS_INDEX,
+} from '../../../../common/endpoint/constants';
+
+export const createActionRequestsEsSearchResultsMock = (): estypes.SearchResponse<
+  EndpointAction | LogsEndpointAction
+> => {
+  const endpointActionGenerator = new EndpointActionGenerator('seed');
+  const fleetActionGenerator = new FleetActionGenerator('seed');
+
+  return endpointActionGenerator.toEsSearchResponse<EndpointAction | LogsEndpointAction>([
+    fleetActionGenerator.generateActionEsHit({
+      action_id: '123',
+      agents: ['agent-a'],
+      '@timestamp': '2022-04-27T16:08:47.449Z',
+    }),
+    endpointActionGenerator.generateActionEsHit({
+      EndpointActions: { action_id: '123' },
+      agent: { id: 'agent-a' },
+      '@timestamp': '2022-04-27T16:08:47.449Z',
+    }),
+  ]);
+};
+
+export const createActionResponsesEsSearchResultsMock = (): estypes.SearchResponse<
+  LogsEndpointActionResponse | EndpointActionResponse
+> => {
+  const endpointActionGenerator = new EndpointActionGenerator('seed');
+  const fleetActionGenerator = new FleetActionGenerator('seed');
+
+  return endpointActionGenerator.toEsSearchResponse<
+    LogsEndpointActionResponse | EndpointActionResponse
+  >([
+    fleetActionGenerator.generateResponseEsHit({
+      action_id: '123',
+      agent_id: 'agent-a',
+      error: '',
```

Comment:
note: If you wish to create successful responses then the `error` key should not be here.

Reply:
Yeah, that would require that the generator be changed. I don't know why it was done that way (to always return an error response), but maybe we change it in the future.